src/olympia/addons/migrations/0036_add_allow_deleted_guid_reuse_switch.py

```diff
@@ -16,6 +16,7 @@ class Migration(migrations.Migration):
 
     dependencies = [
         ('addons', '0035_auto_20210812_1726'),
+        ('waffle', '0001_initial'),
```

Comment:
This is an old migration that created a waffle `Switch`. It previously worked despite not requiring `waffle` because of the way the dependencies resolver ordered the migrations, `waffle` ended up being required first anyway. 

Because the migration I'm adding in `abuse` depends on `addons`, the order ends up being different and could fail if the dependency on `waffle` wasn't added explicitly.

Reply:
This confuses and scares me.

> could fail

Is doing most of the heavy lifting. How did you discover that this _could_ be a problem? How would you have known? Especially if it is not guaranteed to fail, do you just have to model the dependency array like a react hook? Feels very brittle.

Reply:
Tests did fail (for instance https://app.circleci.com/pipelines/github/mozilla/addons-server/29054/workflows/3b52f3cc-d3b1-4bdf-af4b-caf34e58476b/jobs/190061)

I said "could" because it only reproduced on an empty db: on an already existing one, migration engine would only apply the missing migration and we wouldn't have known.